client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoDriver.java

```diff
@@ -1144,4 +1163,26 @@ private static Properties toProperties(Map<String, String> map)
         map.forEach(properties::setProperty);
         return properties;
     }
+
+    public static class TestDnsResolver
+            implements DnsResolver
+    {
+        private final InetAddress address;
+
+        public TestDnsResolver(String context)
+                throws UnknownHostException
+        {
+            this.address = InetAddress.getByName(requireNonNull(context, "context is null"));
```

Comment:
It would be better to do the resolution in the `lookup()` method, since that would be a more typical implementation